Helper/Order.php

```diff
@@ -1644,15 +1644,26 @@ protected function isCaptureHookRequest($newCapture)
 
     /**
      * @param OrderInterface $order
-     * @param                                        $captureAmount
+     * @param float          $captureAmount
      *
+     * @return void
      * @throws \Exception
      */
-    protected function validateCaptureAmount(OrderInterface $order, $captureAmount) {
+    protected function validateCaptureAmount(OrderInterface $order, $captureAmount)
+    {
         $isInvalidAmount = !isset($captureAmount) || !is_numeric($captureAmount) || $captureAmount < 0;
-        $isInvalidAmountRange = $order->getTotalInvoiced() + $captureAmount > $order->getGrandTotal();
 
-        if($isInvalidAmount || $isInvalidAmountRange) {
+        /**
+         * Due to grand total sent to Bolt is rounded, the same operation should be used
+         * when validating captured amount.
+         * Rounding operations are applied to each operand in order to avoid cases when the grand total
+         * is formally less (before it has been rounded) than the sum of the captured amount and the total invoiced.
+         */
+        $isInvalidAmountRange = $this->cartHelper->getRoundAmount($order->getTotalInvoiced())
+            + $this->cartHelper->getRoundAmount($captureAmount)
+            > $this->cartHelper->getRoundAmount($order->getGrandTotal());
+
+        if ($isInvalidAmount || $isInvalidAmountRange) {
             throw new \Exception( __('Capture amount is invalid'));
```

Comment:
can you add amount in error message to make future debugging easier?

Reply:
It's a good point, thank you!